src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -191,6 +191,7 @@ static string NormaliseGeneratedCode(StringWriter writer)
             gennedClass = Regex.Replace(gennedClass, set, " { set; }", RegexOptions.IgnorePatternWhitespace);
             gennedClass = Regex.Replace(gennedClass, @"\s+{\s+}", " { }", RegexOptions.IgnorePatternWhitespace);
             gennedClass = Regex.Replace(gennedClass, @"\)\s+;", ");", RegexOptions.IgnorePatternWhitespace);
+            gennedClass = gennedClass.Replace("()]", "]"); // [Attribute] instead of [Attribute()]
```

Comment:
Is there any need to defend against this replacement being done to a string literal?

Reply:
I'm working on a slightly better way that also takes into account the attribute suffix replacement. I'm thinking about the following regex

```
@"\[(.*)Attribute\((.*)\)\]"
```

and then replace with

```
[$1]
```
if the second matching group is empty. 

What do you think about this?

Reply:
That sounds good. It could still edit string literals, but the chance of it matching is getting smaller and smaller.

> if the second matching group is empty.

Then why not: `@"\[(.*)Attribute\(\)\]"`

Reply:
Looks good.

Reply:
> Then why not: `@"\[(.*)Attribute\(\)\]"`

I wasn't thinking. You need both groups in order to remove `Attribute` when the parens aren't empty.